frontend/catalyst/pennylane_extensions.py

```diff
@@ -982,6 +984,112 @@ def trace_quantum(self, ctx, device, trace, qrp) -> QRegPromise:
         return qrp2
 
 
+class QCtrl(HybridOp):
+    """Catalyst quantum ctrl operation"""
+
+    def _no_binder(self, *_):
+        raise RuntimeError("QCtrl does not support JAX binding")  # pragma: no cover
+
+    binder = _no_binder
+
+    def __init__(
+        self, *args, control_wire_tracers, control_value_tracers, work_wire_tracers, **kwargs
+    ):
+        self.control_wire_tracers: List[Any] = control_wire_tracers
+        self.control_value_tracers: List[Any] = control_value_tracers
+        self.work_wire_tracers: Optional[List[Any]] = work_wire_tracers
+        super().__init__(*args, **kwargs)
+
+    def trace_quantum(self, ctx, device, trace, qrp) -> QRegPromise:
+        raise NotImplementedError("QCtrl does not support JAX quantum tracing")  # pragma: no cover
+
+    def compute_decomposition(self, *params, wires=None, **hyperparameters):
+        """Compute quantum decomposition of the gate by recursively scanning the nested tape and
+        distributing the quantum control operaiton over the tape operations."""
+        assert len(self.regions) == 1, "Qctrl is expected to have one region"
+        assert len(params) == 0, "Decomposition parameters should be empty"
+        assert len(hyperparameters) == 0, "Decomposition hyperparameters should be empty"
+        assert wires is self.wires, "Altering wires is not supported"
+
+        qctrl_check_no_measurements(self.regions[0].quantum_tape)
+        new_tape = qctrl_distribute(
```

Comment:
Not sure it's necessary to check it here again since we already check for measurements in the `ctrl` function.

Reply:
The idea here is to double-check results of user-defined transformations. First check allows to report errors early, the second one makes sure that users did not introduce any measurements while doing transformations. I am open to suggestions though.

Reply:
Oh good point, thanks!